### PR TITLE
test: add case with bad parameters to assert on error message

### LINKED ISSUE #53: Validator does not catch invalid field value in Parameters object
Version: 1.7.0
The following state definition should fail validation:
``` json
{
  "Comment": "This SFN Should Fail Validation but does not",
  "StartAt": "Hello, World",
  "States": {
    "Hello, World": {
      "Type": "Task",
      "Parameters": {
        "foo": "bar",
        "lorem.$": "ipsum",
        "dolor.$": "$.sit.amet"
      },
      "Resource": "arn:aws:states:us-east-1:12345:activity:hello-world",
      "End": true
    }
  }
}
```
` "lorem.$": "ipsum",` is invalid, the value must be a valid JSONPath. Currently it passes validation, but fails on a deploy to AWS.

### PR DESCRIPTION
**Pull request by massfords**
closes #53

### CODE REVIEW

File: src/checks/json-schema-errors.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export default function jsonSchemaErrors(definition: StateMachine): StateMachine
         !instancePathsWithOneOfKeyword.has(error.instancePath))
         .map(error => ({
             'Error code': StateMachineErrorCode.SchemaValidationFailed,
-            Message: error.keyword === "oneOf" ? `${error.instancePath} is invalid.` : error.message as string,
+            Message: `${error.instancePath} is invalid. ${error.message ?? ""}`,
```
Comment (massfords) [inline src/checks/json-schema-errors.ts:83]: changed to always include the instance path in the message

**Comment by massfords** [comment]
@ChristopheBougere my commit message looks to follow the convention. Not sure why it's failing in the build.

File: src/__tests__/error-reporting.test.ts [hunk 2]
```diff
@@ -27,14 +27,28 @@ describe("tests with definitions containing errors to see what's reported", () =
...
+        },
+        {
+            file: "invalid-payload-template.asl.json",
+            expected_errors: [
+                {
+                    "Error code": StateMachineErrorCode.SchemaValidationFailed,
+                    Message: "/States/Hello, World/Parameters/lorem.$ is invalid. must match format \"asl_payload_template\"",
+                    schemaError: {
+                        instancePath: "/States/Hello, World/Parameters/lorem.$",
+                        schemaPath: "#/oneOf/0/patternProperties/%5E.%2B%5C.%5C%24%24/format"
```
Comment (ChristopheBougere) [inline src/__tests__/error-reporting.test.ts:46]: Could you add a comment to explain what’s happening here with %5E.%2B%5C.%5C%24%24 ?

File: src/__tests__/definitions/invalid-payload-template.asl.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "Comment": "This SFN Should Fail Validation but does not",
```
Comment (ChristopheBougere) [inline src/__tests__/definitions/invalid-payload-template.asl.json:2]: Maybe explain why it is failing

File: src/__tests__/error-reporting.test.ts (hunk 2 above)
Comment (massfords) [reply src/__tests__/error-reporting.test.ts:47]: the schema in this case is defined as a patternProperties and the key is a regex that needs escaping as a path.

I left a comment in the src about it. Not sure we need to decode this for display but we could with a url decoder.

File: src/__tests__/definitions/invalid-payload-template.asl.json (hunk 3 above)
Comment (massfords) [reply src/__tests__/definitions/invalid-payload-template.asl.json]: was copy/pasted from the bug report. I updated

File: src/__tests__/error-reporting.test.ts (hunk 2 above)
Comment (ChristopheBougere) [reply src/__tests__/error-reporting.test.ts:47]: Thanks

**Pull request merged**